### PR TITLE
[flang][OpenMP] Add `private` to `allocate` in parallel-sections.f90

### PR DESCRIPTION
**Pull request by kparzysz**
Add a privatizing clause to the construct that uses `allocate` clause. Amend the CHECK lines to reflect the expected output.

### CODE REVIEW

File: flang/test/Lower/OpenMP/parallel-sections.f90 [hunk 1]
```diff
@@ -40,12 +40,8 @@ subroutine omp_parallel_sections_allocate(x, y)
...
   integer, intent(inout) :: x, y
   !CHECK: %[[allocator_1:.*]] = arith.constant 4 : i64
-  !CHECK: %[[allocator_2:.*]] = arith.constant 4 : i64
-  !CHECK: omp.parallel allocate(
-  !CHECK: %[[allocator_2]] : i64 -> %{{.*}} : !fir.ref<i32>) {
-  !CHECK: omp.sections allocate(
-  !CHECK: %[[allocator_1]] : i64 -> %{{.*}} : !fir.ref<i32>) {
-  !$omp parallel sections allocate(omp_high_bw_mem_alloc: x)
+  !CHECK: omp.sections allocate(%[[allocator_1]] : i64 -> %{{.*}} : !fir.ref<i32>) {
+  !$omp parallel sections allocate(omp_high_bw_mem_alloc: x) private(x, y)
```
Comment (kiranchandramohan) [inline flang/test/Lower/OpenMP/parallel-sections.f90:44]: Did you miss the CHECK for parallel?

Comment (kparzysz) [reply flang/test/Lower/OpenMP/parallel-sections.f90:45]: Added.

**Review by tblah — APPROVED** [review]
Thanks for the fix. nit: I think this test also does not need to depend upon openmp_runtime

**Comment by kparzysz** [comment]
> Thanks for the fix. nit: I think this test also does not need to depend upon openmp_runtime

We'd need to remove the `omp_high_bw_mem_alloc` from the `allocate` clause.  That symbol comes from `omp_lib.mod`, which is a part of the runtime.  Are you ok with that, or should we leave it as it is now?

**Review by kiranchandramohan — APPROVED** [review]
LG.

**Comment by tblah** [comment]
Feel free to leave it as it is if it isn't trivial

**Comment by kparzysz** [comment]
The generated code will change slightly.  This isn't the only test where the allocate modifier is used---we can decide later whether we want to keep using it in tests or not.  Better yet, we'd fix the module dependency issue...

**Pull request merged**